url47ebb27197_asirbu_CorrectiveEvolution_tgz-pJ8-eu_fbk_soa_evolution_engine_impl_test_ProblemToSMVFuncTestJ8/src/eu/fbk/soa/util/IOUtils.java

```diff
@@ -0,0 +1,289 @@
+					for (String act : acts) {
+						String[] split3 = act.split(" = ");
+
+						if (split3.length == 2) {
+							String var = split3[0];
+							String value = split3[1].substring(0, split3[1].length()-3);
+							if (value.endsWith("\"")) {
+								value = value.substring(0, value.length()-1);
+							}
+							varMap.clear();
+							varMap.put(var, value);
+
+							Action action = newSTS.addAction(varMap);
+							newSTS.addTransition(new NusmvTransition(fromState, action, toState));
+
+						}
+					}
+				}
+			}
+		}
+
+		return newSTS;
+	}
+	
+	public static STS readSTSFromFile(File file) {
+		String str = IOUtils.readFileAsString(file);	
+		STS newSTS = new DefaultSTS();
+	
+		for (String line : str.split("\n")) {
+			String[] split = line.split("->");
+			if (split.length == 2) {
+				String from = split[0];
+				State fromState = newSTS.getState(from);
+				if (fromState == null) {
+//					System.out.println("Creating state " + from);
+					fromState = new DefaultState(from);
+					newSTS.addState(fromState);
+				}
+
+				String[] split2 = split[1].split("\\[label=\"");
+				if (split2.length == 2) {
+					String to = split2[0];
+
+					State toState = newSTS.getState(to);
+					if (toState == null) {
+						toState = new DefaultState(to);
+						newSTS.addState(toState);
+					}
+					
+					String actName = split2[1].substring(0, split2[1].length() - 4);
+					if (actName.endsWith("\"")) {
+						actName = actName.substring(0, actName.length() - 1);
+					}
+					Action action = new DefaultAction(actName, true);
+					newSTS.addAction(action);
+					newSTS.addTransition(new NusmvTransition(fromState, action, toState));
+				}
+			}
+		}
+		
+		return newSTS;
+	}
+	
+	public static void writeStringToFile(String str, String filePath) {
+		File file = new File(filePath);
+		
+		try {
+			BufferedWriter out = new BufferedWriter(new FileWriter(file, false));
+			out.write(str);
+			out.close();
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+
+	public static void exportSTSToDot(STS sts, String filePath) {
+		File file = new File(filePath);
+		try {
+			IOUtils.exportSTSToDot(sts, new FileWriter(file));
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+	
+	public static void exportSTSToDot(STS sts, Writer writer) {
+		PrintWriter out = new PrintWriter(writer);
+		out.println(sts.toDot());
+		out.flush();
+		out.close();
+	}
+	
+	public static void createImage(String dotFilePath, String outputFilePath) {
+		File dotFile = new File(dotFilePath);
+		File img = new File(outputFilePath);
+		
+		try {
+			String graphvizPath = ConfigUtils.getProperty("graphvizPath");
+			
+			String[] cmdarray = {graphvizPath + "dot.exe", "-Tpng", 
+					dotFile.getAbsolutePath(), "-o", img.getAbsolutePath()};
+			Runtime rt = Runtime.getRuntime();
+			Process p;
+			
+			p = rt.exec(cmdarray);
+			p.waitFor();
+
+		} catch (IOException e1) {
+			e1.printStackTrace();
+		} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)